user-console/src/main/java/org/pentaho/mantle/client/workspace/SchedulesPanel.java

```diff
@@ -121,7 +143,14 @@ public class SchedulesPanel extends SimplePanel {
 
   private ArrayList<IJobFilter> filters = new ArrayList<IJobFilter>();
 
-  private CellTable<JsJob> table = new CellTable<JsJob>( PAGE_SIZE, (CellTableResources) GWT.create( CellTableResources.class ) );
+  private CellTable<JsJob> table = new CellTable<JsJob>( PAGE_SIZE, (CellTableResources) GWT.create( CellTableResources.class ) ) {
+    @Override
+    protected void onBrowserEvent2(Event event) {
+      setFocusedRow( table.getKeyboardSelectedRow() );
+      setFocusedColumn( table.getKeyboardSelectedColumn() );
```

Comment:
Why are the local focused row and focused column necessary?
If needed, are these needed from the outside (public)?
Why is it not enough to call `table.getKeyboardSelectedRow()` and `table.getKeyboardSelectedColumn()` when needed to get the current values?

Reply:
From the above comment.. table.getKeyboardSelectedRow() returns the nextInteractive element's row/column. Hence, we needed a local variable for storing it and later set the focus accordingly.

Reply:
I don't read the screenshot in that way. These two methods return the _currently_ selected cell — the same as the newly created methods. The screenshot shows how these two properties are being updated when certain keys are pressed.

Reply:
`table.getKeyboardSelectedRow()` returns the current row and the screenshot as your rightly understood, shows how the properties are being updated. So, it finds next interactive element and updates the focus to the next available interactive element's row & column. 
So, we are storing them in a focusedRow/focusedColumn variables before the code from the screenshot gets executed and set the focus with the values we have in the focusedRow/focusedColumn variables.

Reply:
> If needed, are these needed from the outside (public)?

removed the getters and setters for `focusedRow` and `focusedColumn`

Reply:
I believe I have the overall picture of this, now.

The new getters and setters, or the underlying fields, should not be needed, and are so in the current implementation because of the way in that the new keyboard handling is being done — it does not override/replace the underlying behavior, and instead, works after it.

Instead of adding a cell preview handler to the table, via `addCellPreviewHandler`, we could/should be using `setKeyboardSelectionHandler`, as is done at `AbstractCellTable#init()`:

```java
setKeyboardSelectionHandler(new CellTableKeyboardSelectionHandler<T>(this));
```

A custom keyboard selection handler would be able to override its `onCellPreview` (listed in the above screenshot), by not calling the base implementation for the specific keys being handled. No local, duplicate storage for the current row and column would be needed.

Reply:
addressed